undertow/src/main/java/org/wildfly/extension/undertow/HostAdd.java

```diff
@@ -76,12 +76,14 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         final String defaultServerName = UndertowRootDefinition.DEFAULT_SERVER.resolveModelAttribute(context, subsystemModel).asString();
         final String defaultHostName = ServerDefinition.DEFAULT_HOST.resolveModelAttribute(context, serverModel).asString();
         final String serverName = serverAddress.getLastElement().getValue();
-
-        boolean isDefaultHost = defaultServerName.equals(serverName) && name.equals(defaultHostName);
+        final boolean isDefaultHost = defaultServerName.equals(serverName) && name.equals(defaultHostName);
+        final ModelNode defaultResponseCode = HostDefinition.DEFAULT_RESPONSE_CODE.resolveModelAttribute(context, model);
```

Comment:
this should be 

final int defaultResponseCode = HostDefinition.DEFAULT_RESPONSE_CODE.resolveModelAttribute(context, model).asInt();
as now there is default set and 404 will be returned if nothing is set.